File: snippets/perl/storageAPI/delete_storage_unit.pl
#Load module netbackup.pm from current directory
use lib"../.";

use gateway;
use storage;
use Getopt::Long qw(GetOptions);
sub printUsage {
  print "\nUsage : perl delete_storage_unit.pl -nbmaster <master_server> -username <username> -password <password> -stu_name <Storage unit name> [-domainname <domain_name>] [-domaintype <domain_type>]\n\n";
  die;
}

my $master_server;
my $username;
my $password;
my $stu_name;
my $domainname;
my $domaintype;

GetOptions(
'nbmaster=s' => \$master_server,
'username=s' => \$username,
'password=s' => \$password,
'stu_name=s'	 => \$stu_name,
'domainname=s' => \$domain_name,
'domaintype=s' => \$domain_type,
) or printUsage();

if (!$master_server || !$username || !$password) {
  printUsage();
}

my $token = gateway::perform_login($master_server, $username, $password, $domain_name, $domain_type);

my $jsonString = storage::delete_storage_unit($master_server, $token, $stu_name);
print "$jsonString\n";

gateway::perform_logout($master_server, $token);

File: snippets/python/storage/delete_storage_unit.py
import sys
import storage
import json
import texttable as tt

protocol = "https"
nbmaster = ""
username = ""
password = ""
domainname = ""
domaintype = ""

port = 1556


def print_usage():
    print("Example:")
    print(
        "python -W ignore delete_storage_unit.py -nbmaster <master_server> -username <username> -password <password> -stu_name <storage unit name>[-domainname <domain_name>] [-domaintype <domain_type>]\n\n\n")


def read_command_line_arguments():
    if len(sys.argv) % 2 == 0:
        print_usage()
        exit()

    global nbmaster
    global username
    global password
    global domainname
    global domaintype
    global stu_name


for i in range(1, len(sys.argv), 2):
    if sys.argv[i] == "-nbmaster":
        nbmaster = sys.argv[i + 1]
    elif sys.argv[i] == "-username":
        username = sys.argv[i + 1]
    elif sys.argv[i] == "-password":
        password = sys.argv[i + 1]
    elif sys.argv[i] == "-stu_name":
        stu_name = sys.argv[i + 1]
    elif sys.argv[i] == "-domainname":
        domainname = sys.argv[i + 1]
    elif sys.argv[i] == "-domaintype":
        domaintype = sys.argv[i + 1]
    else:
        print_usage()
        exit()

if nbmaster == "":
    print("Please provide the value for 'nbmaster'")
    exit()
elif username == "":
    print("Please provide the value for 'username'")
    exit()
elif password == "":
    print("Please provide the value for 'password'")
elif stu_name == "":
    print("Please provide the value for 'stu_name'")
    exit()

print_usage()

read_command_line_arguments()

base_url = protocol + "://" + nbmaster + ":" + str(port) + "/netbackup"

jwt = storage.perform_login(username, password, base_url, domainname, domaintype)

storage.delete_storage_unit(jwt, base_url, stu_name)